src/main/java/randoop/generation/constanttfidf/ConstantMiningStatistics.java

```diff
@@ -0,0 +1,251 @@
+   *     it
+   */
+  public Map<Sequence, Integer> getNumClassesWith(Object scope) {
+    if (GenInputsAbstract.literals_level == ClassLiteralsMode.CLASS) {
+      throw new RandoopBug("Should not get numClassesWith in CLASS level");
+    }
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    if (stats == null) {
+      Log.logPrintf("Scope %s is not a key in scopeStatisticsMap%n", scope);
+      return Collections.emptyMap();
+    }
+    return stats.getNumClassesWith();
+  }
+
+  /**
+   * Get the number of classes in the given scope.
+   *
+   * @param scope a scope
+   * @return the number of classes in the given scope
+   */
+  public Integer getTotalClassesInScope(@Nullable Object scope) {
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    // The default value is null to avoid when scope is java.lang or other standard libraries
+    if (stats == null) {
+      Log.logPrintf("Scope %s is not a key in scopeStatisticsMap%n", scope);
+      return null;
+    }
+    return stats.getNumClasses();
+  }
+
+  /**
+   * Outputs a string representation of the map to the StringBuilder.
+   *
+   * @param   <K2> the type of the map keys
+   * @param   <V2> the type of the map values
+   * @param sb the destination for the string representation
+   * @param indent how many spaces to indent each line of output
+   * @param map the map to print
+   */
+  static <K2 extends @Signed Object, V2 extends @Signed Object> void formatMap(
+      StringBuilder sb, String indent, Map<K2, V2> map) {
+    for (Map.Entry<K2, V2> entry : map.entrySet()) {
+      sb.append(indent);
+      sb.append(entry.getKey());
+      sb.append(" : ");
+      sb.append(entry.getValue());
+      sb.append(System.lineSeparator());
+    }
+  }
+
+  /**
+   * Outputs a string representation of the frequency info to the StringBuilder.
+   *
+   * @param <K1> the type of the outer map keys
+   * @param <K2> the type of the inner map keys
+   * @param <V2> the type of the inner map values
+   * @param sb the destination for the string representation
+   * @param indent how many spaces to indent each line of output
+   * @param header what to print before each inner map
+   * @param mapMap what to print
+   */
+  static <K1 extends @Signed Object, K2 extends @Signed Object, V2 extends @Signed Object>
+      void formatMapMap(
+          StringBuilder sb, String indent, String header, Map<K1, Map<K2, V2>> mapMap) {
+    if (mapMap.isEmpty()) {
+      return;
+    }
+
+    for (Map.Entry<K1, Map<K2, V2>> entry : mapMap.entrySet()) {
+      sb.append(indent);
+      sb.append(header);
+      sb.append(entry.getKey());
+      sb.append(System.lineSeparator());
+      formatMap(sb, indent + "  ", entry.getValue());
+    }
+  }
+
+  /**
+   * Get the scope for the given type based on the literals level.
+   *
+   * @param type the type of the class
+   * @return the scope for the given type
+   */
+  public static Object getScope(@Nullable ClassOrInterfaceType type) {
```

Comment:
Why is the formal parameter `@Nullable`?  I think the formal parameter should be non-null, but the return value should be `@Nullable`.